crates/ibc/src/core/ics04_channel/handler/send_packet.rs

```diff
@@ -94,27 +92,49 @@ pub fn send_packet(
         .into());
     }
 
-    output.log("success: packet send ");
+    Ok(())
+}
+
+/// Per our convention, this message is processed on chain A.
+pub fn send_packet_execute(
+    ctx_a: &mut impl SendPacketExecutionContext,
+    packet: Packet,
+) -> Result<(), ContextError> {
+    {
+        let seq_send_path_on_a = SeqSendPath::new(&packet.port_on_a, &packet.chan_on_a);
+        let next_seq_send_on_a = ctx_a.get_next_sequence_send(&seq_send_path_on_a)?;
+
+        ctx_a.store_next_sequence_send(&seq_send_path_on_a, next_seq_send_on_a.increment())?;
+    }
 
-    let result = SendPacketResult {
-        port_id: packet.port_on_a.clone(),
-        channel_id: packet.chan_on_a.clone(),
-        seq: packet.sequence,
-        seq_number: next_seq_send_on_a.increment(),
-        commitment: ctx_a.packet_commitment(
+    ctx_a.store_packet_commitment(
+        &CommitmentPath {
+            port_id: packet.port_on_a.clone(),
+            channel_id: packet.chan_on_a.clone(),
+            sequence: packet.sequence,
+        },
```

Comment:
Can we rewrite the `CommitmentPath` using the `new()` method?

Reply:
[d07c6e8](https://github.com/cosmos/ibc-rs/pull/468/commits/d07c6e8e2051616ca535e16cfab1ab64d1b5e616)